### ISSUE TITLE
Usage Question about OpenPAI的HA机制

### ISSUE THREAD

**Issue by hellocandy** [microsoft/pai#1970]
**Organization Name**: advantech

**Short summary about the issue/question**:我们打算将Azure上部署的cluster的openPAI部署为HA。有几个问题希望能澄清一下。

**Brief what process you are following**: 
1. Etcd目前同步了哪些东西，仅包括子服务状态，还是子服务的一些数据也包含了？
2. OpenPAI有一个默认的存储路径，这个下面存储的东西在切换M/S node的时候会进行同步吗？
3. HDFS的存储需要进行同步吗？

**How to reproduce it**: 

**OpenPAI Environment**:cluster
- OpenPAI version:0.8.2
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Hardware (e.g. core number, memory size, storage size, GPU type etc.):
- Others:

**Anything else we need to know**:

**Comment by fanyangCS** [comment]
related to https://github.com/Microsoft/pai/issues/852 and https://github.com/Microsoft/pai/issues/724

**Comment by fanyangCS** [comment]
Not sure I fully understand your question. Let me first explain the fault tolerance issue in an Azure deployment. 

If don't care about details, here is the short conclusion: running OpenPAI on Azure can be fault tolerant. But our current released deployment may experience some temporal service availability in case of VM crashes (according to our 1-yr experience, VM crash is a very rare event. You may see at most one VM crash in a month for a cluster with 70 VMs ) 
1.  We enforce etcd replicas for high availability. A single etcd instance crash will not result in any service unavailability)
2. For HDFS, losing single node will not result in any data lost. OpenPAI config multiple replicas for HDFS DataNode. 
3. Sll other components are run in single instance, including key components like WebPortal, Pylon, RestServer, Launcher, Zookeeper, RM, NM, etc. If a VM that contains these components crashed, user will experience service unavailability. But after the VM heals it self (mostly through auto-heal or seldom via manual reboot), the service will resume with no information loss (since we can config the service to keep the data in persistent storage, or the service itself is stateless.) 
4. Any single component failure will not affect existing training jobs, unless the jobs run on the VM that crashed.
5. we plan to improve the service availability by adding replicas for the single-instance components. It will be a gradual process and start from next month.

Hope it addresses your questions.

**Comment by scarlett2018** [comment]
@hellocandy - may you confirm whether @fanyangCS 's answer had fixed your question? thanks.

**Comment by hellocandy** [comment]
Hi,

I create 5 VMs for OpenPAI cluster with HA:
1. Master Node 0
2. Master Node 1
3. Worker Node 0
4. Worker Node 1
5. Proxy Node
And I try to stop all services on Master Node 0(using 'systemctl stop docker'), we lost Port 80 connection on Master Node 1.
Then, I repeat same steps on Master Node 1, the Master Node 0 works fine.
Is there something wrong?

**Comment by hellocandy** [comment]
@fanyangCS Could you please help me to solve this issue? Thanks.

**Comment by fanyangCS** [comment]
@hellocandy, can you explain your question in details? @ydye ,could you follow up this issue?

**Comment by hellocandy** [comment]
Hi,

We expect Master Node 0 to have the same functionality as Master Node 1 to ensure HA. So we tried to shut down services on Master Node 1 (including all components of kubernetes and OpenPAI), at which point pylon (port 80) on Master Node 1 could not be accessed properly. But we tried to stop all services on Master Node 1 (including all components of kubernetes and OpenPAI), at which point Pylon (Port 80) on Master Node 0 can be accessed normally.
